tester/check_replacer.js

```diff
@@ -1,5 +1,100 @@
+/* eslint-disable max-len */
 import {Replacer} from 'transformer/index.js';
-// import {Node, Source} from 'types:assemblyscript/src/ast';
+import {ASTBuilder} from 'assemblyscript';
+
+/**
+ * Generates if expression.
+ *
+ * @param {string} compare
+ * @return {string}
+ */
+function generateIfExpression(compare) {
+  switch (compare) {
+    case ('compare.Equal'):
+      return {'ifExpr': 'got != want', 'hasWant': true};
+    case ('compare.Different'):
+      return {'ifExpr': 'got == want', 'hasWant': true};
+    case ('compare.False'):
+      return {'ifExpr': 'got', 'hasWant': false};
+    case ('compare.True'):
+      return {'ifExpr': '!got', 'hasWant': false};
+    default:
+      return {'ifExpr': compare, 'hasWant': compare.search('want')>-1};
+  }
+}
+
+/**
+ * Generates a test.
+ *
+ * There is two test models:
+ * - When the result value is a boolean:
+ *   - in that case we don't need to:
+ *     - calculate a wanted value,
+ *     - add the expecting value in the error message.
+ * - Otherwise we need a wanted value.
+ *
+ * @param {string} name - test friendly name
+ * @param {string} got - got expression
+ * @param {string} ifExpression
+ * @param {string} needWant - include want in test ?
+ * @param {bool} continueOnFailure - should the test failure stop the test suite
+ * @param {Array} value - array of values to populate got template with
+ * @param {number} iValue - current index of values
+ * @return {Object}
+ */
+function generateTest(name, got, ifExpression, needWant, continueOnFailure, value, iValue) {
+  let expr;
+  if (needWant) {
+    expr = `
+    test('${name}', () => {
+      const got = ${got};
+      const want = ${ASTBuilder.build(value[iValue])};
```

Comment:
Why do we need to "build" the array?

Reply:
Because the value here is a [node](https://github.com/AssemblyScript/assemblyscript/blob/a7d10ba0c50caf0be8b29dcb4c71d378b0d05dc6/src/ast.ts#L129) which is super complex to reverse to simple text.

Now double thinking about it, I should maybe extract the value from the string content...

Reply:
I finally simplified the code and remove ASTBuilder. Tell me if its clearer !

Reply:
It is !